robota_core/tests/test_artefact_urls.py

```diff
@@ -0,0 +1,16 @@
+import pytest
+from robota_core.repository import GitlabRepository, GithubRepository
+
+class TestGithubUrls:
+    repo = "https://github.com/robota-suite/robota-core"
```

Comment:
It was clever to use the robota-core repository itself for this as we can guarantee that it exists and has the branch named.  Your approach made me think about how we should approach this kind of testing - testing again real repo data rather than just whipping up the dicts in the test.  You're right that we should do some acceptance testing against real repos.

So I made a couple of test repos for this, one in GitHub and one in GitLab:

https://github.com/robota-suite/test-public-repo

https://gitlab.com/robota-suite/test-public-repo

I suggest we leave this PR as it is, but then create an issue (or issues) for moving tests like this onto the
more stable test repos.  I think this should give good test coverage alongside the current dict-based tests
that we use now.